src/main/java/com/google/sps/data/PersistentMatchRepository.java

```diff
@@ -87,8 +92,14 @@ public void removeMatch(User user, User match) throws Exception {
     }
 
     for(String userId : matches) {
-      User newMatch = userRepository.fetchUserWithId(userId);
-      toReturn.add(newMatch);
+      try {
+        User newMatch = userRepository.fetchUserWithId(userId);
+        toReturn.add(newMatch);
+      //catch exception if user is not in persistent repository
+      } catch (Exception e) {
+          System.err.println("UserID was not found in Persistent Repository: " + userId);
```

Comment:
nit: I like the extra error context! However this might also fail if datastore's unreachable, bad userId, etc. Maybe append the error to your print?
`System.err.println("UserID was not found in Persistent Repository: " + userId +  ". Error: " + e.toString());`

Reply:
Done!